src/Compilers/CSharp/Test/Semantic/Semantics/StructConstructorTests.cs

```diff
@@ -0,0 +1,1329 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Microsoft.CodeAnalysis.CSharp.Test.Utilities;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics
+{
+    public class StructConstructorTests : CSharpTestBase
+    {
+        [CombinatorialData]
+        [Theory]
+        public void PublicParameterlessConstructor(bool useCompilationReference)
+        {
+            var sourceA =
+@"public struct S
+{
+    public readonly bool Initialized;
+    public S() { Initialized = true; }
+}";
+            var comp = CreateCompilation(sourceA, parseOptions: TestOptions.Regular9);
+            comp.VerifyDiagnostics(
+                // (4,12): error CS8652: The feature 'parameterless struct constructors' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                //     public S() { Initialized = true; }
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "S").WithArguments("parameterless struct constructors").WithLocation(4, 12));
+
+            comp = CreateCompilation(sourceA, parseOptions: TestOptions.RegularPreview);
+            comp.VerifyDiagnostics();
+            var refA = AsReference(comp, useCompilationReference);
+
+            var sourceB =
+@"using System;
+class Program
+{
+    static T CreateNew<T>() where T : new() => new T();
+    static T CreateStruct<T>() where T : struct => new T();
+    static void Main()
+    {
+        Console.WriteLine(new S().Initialized);
+        Console.WriteLine(CreateNew<S>().Initialized);
+        Console.WriteLine(CreateStruct<S>().Initialized);
+    }
+}";
+            CompileAndVerify(sourceB, references: new[] { refA }, expectedOutput:
+$@"True
+True
+{ExecutionConditionUtil.IsCoreClr}"); // Desktop framework ignores constructor in Activator.CreateInstance<T>() where T : struct.
```

Comment:
Wasn't this desktop issue fixed a long time ago?

Reply:
The parameterless constructor is not invoked in the `where T : struct` case.

Reply:
I'd missed that Core vs. desktop difference.
Makes me wonder, should we do a runtime capability check (like DIM and some other features?)

Reply:
Added a note to test plan